module/executiondatasync/pruner/pruner.go

```diff
@@ -169,9 +168,16 @@ func (p *Pruner) loop(ctx irrecoverable.SignalerContext, ready component.ReadyFu
 		select {
 		case <-ctx.Done():
 			return
-		case height := <-p.fulfilledHeights:
-			if height > p.lastFulfilledHeight {
-				p.lastFulfilledHeight = height
+		case <-p.fulfilledHeight.Channel():
```

Comment:
I think we should refactor the existing logic a bit to make this simpler.

Instead of all data coming into the loop over channels, let's make the config values `heightRangeTarget` and `threshold` atomics, and set their values directly in the setters.

Then, update the loop to trigger based on a timer. We can just use some default value like 10 minutes or something like that.

That allows you to remove the notifier from the producers, which should simplify the logic here.

Reply:
[04a0a73](https://github.com/onflow/flow-go/pull/6217/commits/04a0a73186d4447615ab71e3b4830e891781616d)